site/en/blog/immutable-document-domain/index.md

```diff
@@ -0,0 +1,197 @@
+---
+layout: 'layouts/blog-post.njk'
+title: "`document.domain` is becoming immutable"
+description: >
+  Websites will be unable to set `document.domain` on Chrome starting in version 101. If your website relies on setting `document.domain`, your action is required.
+subhead: >
+  Beginning with Chrome version 101, websites will be unable to set `document.domain`.
+date: 2022-01-06
+authors:
+  - agektmr
+tags:
+  - security
+hero: 'image/YLflGBAPWecgtKJLqCJHSzHqe2J2/grGMzuhOjsGhN150dONe.jpg'
+alt: >
+  A dog in disguise.
+---
+
+**If your website relies on relaxing the same-origin policy via `document.domain`,
+your action is required.**
+
+Beginning with Chrome version 101, websites will be unable to set
+`document.domain`. Websites will need to use alternative approaches such as
+`postMessage()` or Channel Messaging API to communicate cross-origin. If a
+website relies on same-origin policy relaxation via `document.domain` to
+function correctly, it will need to send an `Origin-Agent-Cluster: ?0` header
+along with all documents that require that behavior (note that `document.domain`
+has no effect if only one document sets it).
+
+## Why make `document.domain` immutable?
+
+Many websites set `document.domain` to allow communication between [same-site
+but cross-origin](https://web.dev/same-site-same-origin/) pages. 
+
+{% Aside 'key-term' %}
+
+Sites which have the same
+[eTLD+1](https://web.dev/same-site-same-origin/#:~:text=the%20whole%20site%20name%20is%20known%20as%20the%20etld%2B1)
+but are different subdomains are considered "same-site but cross-origin".
+
+{% endAside %}
+
+Here's how it's used:
+
+Let's say a page on `https://parent.example.com` embeds an iframe page from
+`https://video.example.com`. These pages have the same eTLD+1 (`example.com`)
+with different subdomains. By setting both pages' `document.domain` to
+`'example.com'`, the two origins can be treated as if they are the same-origin
+by the browser.
+
+On `https://parent.example.com`:
```

Comment:
It would be good to spell out what the code snippets are demonstrating. Non-blocking comment, but it's a good practice. Even if the code seems simple, you're giving readers a heads up on what comes next.

Reply:
I believe the previous paragraph and the inline comments do that job.

Reply:
I think it could be cleaned up a little further. Added clearer code comments and confirmed purpose beforehand